src/drivers/intel/cavs/idc.c

```diff
@@ -161,6 +161,37 @@ static int idc_msg_status_get(uint32_t core)
 	return status;
 }
 
+/**
+ * \brief Waits until slave core receives the message.
+ * \param[in] core Id of the core receiving the message.
+ * \return Error code.
+ */
+static int idc_wait_in_blocking_mode(uint32_t core)
+{
+	struct timer *timer = timer_get();
+	struct idc *idc = *idc_get();
+	uint64_t deadline;
+
+	deadline = platform_timer_get(timer) +
+		clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK, 1) *
+		IDC_TIMEOUT / 1000;
+
+	while (!idc->msg_processed[core]) {
```

Comment:
@lyakh Isn't shared memory allocated from an uncached region on CAVS (the only platform currently sporting multicore)?

Reply:
@paulstelian97 is right. It's uncached do we don't need to do anything.

Reply:
@tlauda ok, my confusion largely arises from the presence and what seems to me an often wrong use of the `platform_shared_commit()` dummy function. It is a NOP so, calling it anywhere an arbitrary number of times doesn't change anything, expect creating confusion :-)

Reply:
platform_shared_commit() is to be called whenever you're done working with a shared memory region. Theoretically on other platforms it would do a writeback_invalidate.

Reply:
@paulstelian97 ok, then this should be done here too?

Reply:
Here in particular we're doing reads and thus platform_shared_commit shall _not_ be called. It is to be called only on writes.

Reply:
@paulstelian97 no, that is wrong. Above you yourself said "whenever you're done working with a shared memory region." And the comment to that function says: "It's used after usage of data shared by different cores." And it's also logical: with this API caches are flushed *after* reading the data to make sure that data is *never* in cache between uses. That way next time you come to read that data on the same core you don't need to invalidate the cache - you know, that that data isn't cached. If you don't flush caches after nor before reading, you'll get stale data in cache and miss another core modifying that data.

Reply:
@lyakh @paulstelian97 Actually idc struct is not shared. It's allocated per core. msg_processed is set in irq_handler. It looks like it should be replaced by IDC registers checking instead. I will do it in another patch.

Reply:
@tlauda interesting... Then why did your patches work?? Weird.

Reply:
Why HW busy/done bits are not used to sync IDC messaging? Extra things like 'msg_processed' would not be needed.

Reply:
it seems it's different: in the ISR idc is the argument. And ISRs are registered globally, so, whenever an interrupt occurs it's delivered to all cores, right? So, you get idc from a different core...